internal/broker/broker.go

```diff
@@ -838,6 +838,19 @@ func (b *Broker) refreshToken(ctx context.Context, oauth2Config oauth2.Config, o
 	return t, nil
 }
 
+func (b *Broker) refreshUserInfo(ctx context.Context, session *session, t *token.AuthCachedInfo, oldUserInfo info.User) (info.User, error) {
+	userInfo, err := b.fetchUserInfo(ctx, session, t)
+	if err != nil {
+		return info.User{}, fmt.Errorf("could not refresh user info: %w", err)
+	}
+
+	// Don't overwrite the shell and home directory, because those should be configurable locally.
+	userInfo.Shell = oldUserInfo.Shell
+	userInfo.Home = oldUserInfo.Home
```

Comment:
This works fine with the case in which the home changes (gets moved) because the broker prefix has?